metadata_layer.py

```diff
@@ -606,14 +606,22 @@ def links(self, arg_links):
                     rights_uri = link.rights_uri or self.default_rights_uri
                     if open_access_link and not rights_uri in RightsStatus.OPEN_ACCESS:
                         rights_uri = RightsStatus.GENERIC_OPEN_ACCESS
-                    self.formats.append(
-                        FormatData(
-                            content_type=link.media_type,
-                            drm_scheme=DeliveryMechanism.NO_DRM,
-                            link=link,
-                            rights_uri=rights_uri,
+                    format_found = False
+                    for format in self.formats:
+                        if format.link and format.link.href == link.href:
+                            if not format.rights_uri:
+                                format.rights_uri = rights_uri
+                            format_found = True
+                            break
+                    if not format_found:
+                        self.formats.append(
+                            FormatData(
+                                content_type=link.media_type,
+                                drm_scheme=DeliveryMechanism.NO_DRM,
+                                link=link,
+                                rights_uri=rights_uri,
+                            )
```

Comment:
There are some places in the content server where we explicitly create a format for an open access link. If a format for the link was already passed in, we don't need to create a new one.